workshops/web_chat/README.md

```diff
@@ -0,0 +1,782 @@
+    font-family: sans-serif;
+    padding: 5px;
+    margin: 0px;
+    border: 0px;
+}
+
+b {
+    padding: 0px;
+}
+
+body,
+html {
+    width: 100%;
+    height: 100%;
+}
+
+#messageLog {
+    width: 100%;
+    height: 90%;
+    overflow: auto;
+}
+
+#controls {
+    width: 100%;
+    height: 10%;
+    display: grid;
+    grid-template-columns: 2fr 7fr 1fr;
+    grid-gap: 10px;
+}
+
+#name {
+    height: 100%;
+    width: 100%;
+    border: 1px solid gray;
+    border-radius: 5px;
+    text-align: center;
+    word-wrap: break-word;
+}
+
+#message {
+    height: 100%;
+    width: 100%;
+    border: 1px solid gray;
+    border-radius: 5px;
+}
+
+#send {
+    height: 100%;
+    width: 100%;
+    border-radius: 5px;
+}
+```
+
+Explanation:
+  - Here, we're adding CSS to make our HTML look like our prototype
+    - `<body>` is made to take up the full page
+    - The message box takes up 90% of body's height
+    - The controls take up 10% of body's height
+    - `<div id="controls">` is a grid, where 
+      - 2/10 of the space is taken up by the name
+      - 7/10 of the space is taken up by the message box
+      - 1/10 of the space is taken up by the button 
+
+
+
+<details>
+    <summary>This is how the HTML should look:</summary>
+
+```html
+<!DOCTYPE html>
+<html>
+
+<head>
+    <style>
+        * {
+            box-sizing: border-box;
+            font-family: sans-serif;
+            padding: 5px;
+            margin: 0px;
+            border: 0px;
+        }
+
+        b {
+            padding: 0px;
+        }
+
+        body,
+        html {
+            width: 100%;
+            height: 100%;
+        }
+
+        #messageLog {
+            width: 100%;
+            height: 90%;
+            overflow: auto;
+        }
+
+        #controls {
+            width: 100%;
+            height: 10%;
+            display: grid;
+            grid-template-columns: 2fr 7fr 1fr;
+            grid-gap: 10px;
+        }
+
+        #name {
+            height: 100%;
+            width: 100%;
+            border: 1px solid gray;
+            border-radius: 5px;
+            text-align: center;
+            word-wrap: break-word;
+        }
+
+        #message {
+            height: 100%;
+            width: 100%;
+            border: 1px solid gray;
+            border-radius: 5px;
+        }
+
+        #send {
+            height: 100%;
+            width: 100%;
+            border-radius: 5px;
+        }
+    </style>
+</head>
+
+<body>
+    <div id="messageLog"></div>
+    <div id="controls">
+        <input id="name" placeholder="Your name">
+        <textarea id="message" placeholder="Type your message here"></textarea>
+        <button id="send">
+            Send
+        </button>
+    </div>
+</body>
+</html>
+```
+
+</details>
+
+### Adding Javascript
+
+- Create a new file called frontend.js - This will be the code that runs on someone's device when they visit your webpage
+- Add a `<script src="frontend.js"></script>` just before body ends
```

Comment:
Show a code block with this in the html file in context, so that people know where to put this

Reply:
I'm confused here, there's already a `<details>` containing the complete HTML file after this